tests/func/test_commit.py

```diff
@@ -1,46 +1,48 @@
 from __future__ import unicode_literals
 
+import os
+
 import pytest
 
 from dvc.stage import StageCommitError
 from dvc.utils.stage import dump_stage_file
 from dvc.utils.stage import load_stage_file
 
 
-def test_commit_recursive(dvc_repo, repo_dir):
-    stages = dvc_repo.add(repo_dir.DATA_DIR, recursive=True, no_commit=True)
-    assert len(stages) == 2
-
-    assert dvc_repo.status() != {}
+def test_commit_recursive(tmp_dir, dvc):
+    tmp_dir.gen(
+        {"dir": {"file": "file content", "subdir": {"file2": "file2 content"}}}
```

Comment:
You can use simply `{"file1": "text1", ...}`, or even `{"foo": "foo"}` if you are tying to make this shorter. Up to you.

Reply:
Kudos for not using `repo_template`)